### PR TITLE
Test ignoring unexpected records

### PR DESCRIPTION
**Pull request by andrewpollock**
Addresses #2433

### CODE REVIEW

File: source_test.yaml [hunk 1]
```diff
@@ -44,6 +44,7 @@
 - name: 'android'
   versions_from_repo: False
   type: 1
+  ignore_patterns: ['^(?!A-).*$']
```
Comment (oliverchang) [inline source_test.yaml:47]: I don't think this one is correct. 

There may be other cases for us to consider. They also export PUB-A-*.json for example.

**Review by oliverchang — APPROVED** [review]

**Comment by andrewpollock** [comment]
I haven't desk-checked this against https://github.com/google/osv.dev/blob/f182e67869f4963d1cfc406b75953edd46eb6338/osv/models.py#L836-L846 yet

File: source_test.yaml (hunk 1 above)
Comment (andrewpollock) [reply source_test.yaml]: Good catch, thanks. Corrected in 203fec69

**Review by oliverchang — APPROVED** [review]

**Comment by andrewpollock** [comment]
I'm feeling fairly confident about this approach now that I've added some test coverage in aaa03831

File: source_test.yaml [hunk 2]
```diff
@@ -68,6 +70,7 @@
   versions_from_repo: False
   rest_api_url: 'https://curl.se/docs/vuln.json'
   type: 2
+  ignore_patterns: ['^(?!CURL-).*$']
```
Comment (oliverchang) [inline source_test.yaml:73]: actually, I just noticed this line is duplicated with line 74. It also isn't applicable for REST sources right?

**Review by oliverchang — APPROVED** [review]

File: source_test.yaml (hunk 2 above)
Comment (andrewpollock) [reply source_test.yaml]: Thanks for catching that. Fixed in 8c34b659, and yes, by the looks of things, https://github.com/google/osv.dev/blob/f182e67869f4963d1cfc406b75953edd46eb6338/docker/importer/importer.py#L58 is not currently called from https://github.com/google/osv.dev/blob/f182e67869f4963d1cfc406b75953edd46eb6338/docker/importer/importer.py#L652

I've left the entries for the two REST sources as a statement of intent, and added a comment to make it obvious that it's not currently honoured.

**Review by oliverchang — APPROVED** [review]

**Pull request merged**